File: server/src/test/java/com/pass/unit/exam/domain/QuestionTypeTest.java
package com.pass.unit.exam.domain;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

import com.pass.exam.domain.QuestionType;
import com.pass.global.exception.NotFoundException;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class QuestionTypeTest {

    @Test
    @DisplayName("질문 유형 이름으로 질문 유형을 찾을 수 있다.")
    void from() {
        // given
        String questionTypeName = "SHORT_ANSWER";

        // when
        QuestionType foundQuestionType = QuestionType.from(questionTypeName);

        // then
        assertThat(foundQuestionType).isEqualTo(QuestionType.SHORT_ANSWER);
    }

    @Test
    @DisplayName("질문 유형 이름으로 질문 유형을 찾을 수 없을 경우 예외를 발생시킨다.")
    void fromFailWhenQuestionTypeNotFound() {
        // given
        String questionTypeName = "NOT_EXIST";

        // when & then
        assertThatThrownBy(() -> QuestionType.from(questionTypeName))
                .isInstanceOf(NotFoundException.class)
                .hasMessage("존재하지 않는 질문 유형입니다. 질문 유형: NOT_EXIST");
    }
}

File: server/src/test/kotlin/com/fluffy/exam/domain/QuestionTypeTest.kt
package com.fluffy.exam.domain

import com.fluffy.global.exception.NotFoundException
import io.kotest.assertions.throwables.shouldThrow
import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe

class QuestionTypeTest : StringSpec({

    "존재하는 문제 유형은 정상적으로 반환된다." {
        // given
        val value = "SINGLE_CHOICE"

        // when
        val questionType = QuestionType.from(value)

        // then
        questionType shouldBe QuestionType.SINGLE_CHOICE
    }

    "대소문자 구분 없이 정상적으로 반환된다." {
        // given
        val value = "true_or_false"

        // when
        val questionType = QuestionType.from(value)

        // then
        questionType shouldBe QuestionType.TRUE_OR_FALSE
    }

    "존재하지 않는 문제 유형은 예외를 던진다." {
        // given
        val invalidValue = "INVALID_QUESTION_TYPE"

        // when & then
        shouldThrow<NotFoundException> { QuestionType.from(invalidValue) }
            .message shouldBe "존재하지 않는 문제 유형입니다. 문제 유형: $invalidValue"
    }
})
